com.unity.netcode.gameobjects/Tests/Runtime/MultiInstanceHelpers.cs

```diff
@@ -17,10 +17,80 @@ public static class MultiInstanceHelpers
         public const int DefaultMinFrames = 1;
         public const int DefaultMaxFrames = 64;
         private static List<NetworkManager> s_NetworkManagerInstances = new List<NetworkManager>();
+        private static Dictionary<NetworkManager, MultiInstanceHooks> s_Hooks = new Dictionary<NetworkManager, MultiInstanceHooks>();
         private static bool s_IsStarted;
         private static int s_ClientCount;
         private static int s_OriginalTargetFrameRate = -1;
 
+        public delegate bool MessageReceiptCheck(object receivedMessage);
+
+        private class MultiInstanceHooks : INetworkHooks
+        {
+            public bool IsWaiting;
+
+            public MessageReceiptCheck ReceiptCheck;
+
+            public static bool CheckForMessageOfType<T>(object receivedMessage) where T : INetworkMessage
+            {
+                return receivedMessage is T;
+            }
+
+
+            public void OnBeforeSendMessage<T>(ulong clientId, ref T message, NetworkDelivery delivery) where T : INetworkMessage
```

Comment:
why these method bodies are empty?

Reply:
Nothing to do in them here, but required by the interface.

Reply:
I think the interface should have been a base class so it could provide empty default implementations. But that change is out of scope for this PR, I think.

Reply:
so, do you have any plans/tickets/PRs to move these under a base class like `BaseMultiInstanceTest`?

Reply:
I'll do the refactor as soon as this branch gets merged in, otherwise it might create merge conflicts in this branch that git won't detect.

Reply:
would that be a big or unrelated PR? I'm asking becase this PR could also have the refactors too, no?

Reply:
It wouldn't be that big but it does affect things outside this PR's scope. I don't want to try to "sneak in" that kind of refactor in unrelated work.

Reply:
sure.